synapse/common.py

```diff
@@ -658,14 +659,23 @@ def setlogging(mlogger, defval=None):
     Returns:
         None
     '''
+    log_struct = envbool('SYN_LOG_STRUCT', 'false')
```

Comment:
currently our logging is orthogonal to cell configuration; and done right away during application startup. The current state of this retains that PR, since moving it to a cell configuration directive introduces a startup race where we don't know the final logging configuration until we are inside of the cell __anit__ routine.

Reply:
Could we maybe do this detection/initialization within the synapse/servers/foo.py / synapse/tools/foo.py to allow them to be a bit more concrete?

Reply:
I've rewritten this to add it as a option in cell.initFromArgv; so we have the option of a command line switch or environment variable. This is then propagated as an opaque structure to places where we use it for subprocesses.